### PR TITLE
DM-30718: Do not recreate DataCoordinate on butler.get if given a DataCoordinate

### PR DESCRIPTION
**Pull request by timj**
Previously a fully expanded DataCoordinate would be turned into a minimalist dict before converting it back to a DataCoordinate in an attempt to handle the detector/exposure record specifications in dataIds.

## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/_butler.py [hunk 1]
```diff
@@ -533,56 +533,52 @@ def _standardizeArgs(self, datasetRefOrType: Union[DatasetRef, DatasetType, str]
...
-            Raised if a resolved `DatasetRef` was passed as an input, but it
-            differs from the one found in the registry.
-        TypeError
-            Raised if no collections were provided.
+        dataId : `dict` or `DataCoordinate`
+            The, possibly rewritten, dataId. If given a `DataCoordinate` and
+            no keyword arguments, the orginal dataId will be returned
+            unchanged.
+        kwds : `dict`
+            Any unused keyword arguments.
```
Comment (TallJimbo) [inline python/lsst/daf/butler/_butler.py:577]: I've been told to switch to `kwargs` in new code, and I think we usually put the leading `**` in the docs now, too.

File: python/lsst/daf/butler/_butler.py [hunk 2]
```diff
@@ -776,6 +767,61 @@ def quote(s: Any) -> str:
...
+        collections : Any, optional
+            Collections to be searched, overriding ``self.collections``.
+            Can be any of the types supported by the ``collections`` argument
+            to butler construction.
+        allowUnresolved : `bool`, optional
+            If `True`, return an unresolved `DatasetRef` if finding a resolved
+            one in the `Registry` fails.  Defaults to `False`.
+        kwds
+            Additional keyword arguments used to augment or construct a
+            `DataId`.  See `DataId` parameters.
```
Comment (TallJimbo) [inline python/lsst/daf/butler/_butler.py:798]: Same doc comments as above.

**Review by TallJimbo — APPROVED** [review]
Looks good. Only comments are about fixing doc style issues that are probably my fault originally, since you're touching those lines anyway (or at least `git diff` thinks you are).

File: python/lsst/daf/butler/_butler.py (hunk 1 above)
Comment (timj) [reply python/lsst/daf/butler/_butler.py:577]: Yes, this was your code moved around. Will change it.

**Pull request merged**